java/ql/src/experimental/Security/CWE/CWE-522/InsecureBasicAuth.ql

```diff
@@ -88,7 +87,7 @@ class HttpStringLiteral extends StringLiteral {
   HttpStringLiteral() {
     // Match URLs with the HTTP protocol and without private IP addresses to reduce false positives.
     exists(string s | this.getRepresentedString() = s |
-      s.regexpMatch("(?i)http://[\\[a-zA-Z0-9].*") and
+      s.regexpMatch("(?i)http://[\\[:a-zA-Z0-9].*") and
```

Comment:
Maybe this should be reverted because it would now match `new URL("http://:8000")`. My comments regarding special handling for empty hosts in `getPrivateHostRegex` might have been incorrect. The previous version (without colon) would not have matched these empty host URLs as desired.

Why did you change this regex to match the colon now?

Reply:
Sorry I confused myself. Non-local IPv6 addresses start with a number so colon is not needed. I've removed the added colon.

Now the query works with empty host loopback address no matter it is in a concatenated string or a separate string/variable. 

The code seems to work beautifully with all test scenarios. Please let me know if I still miss something. I won't mark this conversation as resolved until your confirmation. Thanks @Marcono1234 for all your help.